src/mongocxx/options/data_key.hpp

```diff
@@ -30,6 +30,8 @@ class client_encryption;
 
 namespace options {
 
+using key_material_type = std::vector<uint8_t>;
```

Comment:
```suggestion
using key_material_type = std::array<uint8_t, 96>;
```
Suggest non-allocating and fixed-length type for `key_material_type` (key material length is [specified by spec](https://github.com/mongodb/specifications/blob/baf3724155500acbb557e158b8a5d4334e7f3512/source/client-side-encryption/client-side-encryption.rst#keymaterial) to be exactly 96 bytes).

Also suggest moving type alias definition into `data_key` class for consistency with other `options` classes such as: https://github.com/mongodb/mongo-cxx-driver/blob/57e37617421a8cb7210856ca491ce2131d9b4127/src/mongocxx/options/client_encryption.hpp#L77

CC @kevinAlbs I think we discussed this before but I do not remember the details. Is this suggestion appropriate, or did we conclude that we want to leave the length open to change in the future (in which case `std::vector` may be appropriate)?

Reply:
My understanding of that conversation was the conclusion was to make the key material type variable and leave all validation of length to libmongocrypt. Thread here: https://mongodb.slack.com/archives/GMY98JS86/p1677166844729879?thread_ts=1677164857.739539&cid=GMY98JS86